dashboard/lib/services/lti.rb

```diff
@@ -176,14 +176,18 @@ def self.sync_course_roster(lti_integration:, lti_course:, nrps_sections:, secti
       if lti_section.nil?
         section = Section.new(
           {
-            user_id: section_owner_id,
+            user_id: current_user_id,
             name: section_name,
             login_type: Section::LOGIN_TYPE_LTI_V1,
           }
         )
         lti_section = LtiSection.create(lti_course_id: lti_course.id, lms_section_id: lms_section_id, section: section)
         had_changes = true
       end
+
+      # Only sync the section if the section's owner is equal to the current user
+      next unless lti_section.section.user_id == current_user_id
```

Comment:
Should this check be moved earlier in the function? It seems like after line 162 where we get the array of sections, we could check the owner of the first section and short-circuit there if we have a mismatch.